bibliopixel/drivers/SPI/driver.py

```diff
@@ -0,0 +1,42 @@
+from .. channel_order import ChannelOrder
+from .. driver_base import DriverBase
+from ... util.enum import resolve_enum
+from . import interfaces
+from enum import IntEnum
+# SubDriver imports
+from . APA102 import APA102
+from . LPD8806 import LPD8806
+from . WS281X import WS281XSPI
+from . WS2801 import WS2801
+
+
+class SPI_LEDTYPE(IntEnum):
+    APA102 = 0
+    DOTSTAR = 0
+    WS2801 = 1
+    WS281X = 2
+    WS2812 = 2
+    WS2812B = 2
+    NEOPIXEL = 2
+    LPD8806 = 3
+
+
+SPI_DRIVERS = [
+    APA102,
+    WS2801,
+    WS281XSPI,
+    LPD8806
+]
+
+
+class SPI(DriverBase):
```

Comment:
I don't see how this is relevant!

You get the _same_ result from either code section.  Any reflection operation that works on the weird metaclass, works exactly the same on the plain old class with the boring construction function.

Of course we're getting rid of `MANIFEST` but the same reflection operations will work on both "weird metaclass" and "regular old function".

As I said, I'll instantly drop this objection _if_ you can give me a code snippet or even pseudo-code which would be different between the two strategies.

[Doe normaal!!](https://stuffdutchpeoplelike.com/2010/11/26/no-56-normalcy-doe-normaal/) :-P

Thing is - there is _one_ good use for metaclasses and that's "aspect-oriented programming" - where you _rewrite_ the derived class in order to, e.g. log before and after each method call.

That's a very good use, but this isn't it.  You aren't rewriting the class, so there isn't any need for a metaclass.